### PR TITLE
Add address bar to the Webbrowser [DISCL-380]

### PR DESCRIPTION
**Pull request by rdumusc**
needs https://github.com/BlueBrain/Deflect/pull/119

### CODE REVIEW

File: tide/core/PixelStreamInteractionDelegate.h [hunk 1]
```diff
@@ -70,7 +70,7 @@ class PixelStreamInteractionDelegate : public ContentInteractionDelegate
     void swipeDown() override;
     //@}
 
-    /** @name Mouse gesture handlers. */
+    /** @name Keyboard gesture handlers. */
```
Comment (tribal-tec) [inline tide/core/PixelStreamInteractionDelegate.h]: keyboard has gestures??

Comment (rdumusc) [inline tide/core/PixelStreamInteractionDelegate.h]: haha :-)

File: tide/core/WebbrowserContent.cpp [hunk 2]
```diff
@@ -58,12 +65,113 @@ bool WebbrowserContent::hasFixedAspectRatio() const
...
+        return;
+
+    _addressBarFocused = set;
+    emit addressBarFocusedChanged();
+    emit modified();
+}
+
+void WebbrowserContent::parseData( const QByteArray data )
+{
+    std::istringstream iss{ data.toStdString(), std::istringstream::binary };
```
Comment (tribal-tec) [inline tide/core/WebbrowserContent.cpp]: group the de- and serialization for the webbrowser together? It's in two different files now.

Comment (rdumusc) [inline tide/core/WebbrowserContent.cpp]: Right, and at some point when testing I forgot to update one of the two sides... ;-) I'll change that

**Comment by tribal-tec** [comment]
Adding a test of the webbrowser serialization and the RESTServer?

**Comment by rdumusc** [comment]
Done, with additonal cleanups

File: tide/master/rest/RestServer.cpp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+RestServer::RestServer( const int port )
+    : _httpServer{ zeroeq::URI { QString(":%1").arg( port ).toStdString( )}}
+{
+    _socketNotifier.connect( &_socketNotifier, &QSocketNotifier::activated,
+                             [this]() { _httpServer.receive( TIMEOUT ); });
+}
+
+int RestServer::getPort() const
+{
+    return _httpServer.getURI().getPort();
```
Comment (tribal-tec) [inline tide/master/rest/RestServer.cpp]: not only getPort()?

**Comment by tribal-tec** [comment]
+2 otherwise. Good job!

File: tide/master/rest/RestServer.cpp (hunk 3 above)
Comment (rdumusc) [inline tide/master/rest/RestServer.cpp]: I'm not seeing any such method, I think it has to be retrieved from he uri...

**Pull request merged**